rock tries to resolve enum values inside versioned-out blocks

```ooc
version(windows) {
include windows
TestEnum: enum {
    Symbol1 = WAIT_OBJECT_0
}
}
```

this won't compile on Unix as rock tries to resolve the value of Symbol1, even though everything is inside version block specific to Windows:


> bin/rock --version
> rock 0.9.11-head codename sapporo, built on Mon Oct 19 14:24:24 2015

> rock_tmp/ooc/enums/enums.c: In function ‘enums_load’:
> rock_tmp/ooc/enums/enums.c:98:13: error: ‘WAIT_OBJECT_0’ undeclared (first use in this function)
>             WAIT_OBJECT_0

--- a/source/rock/middle/EnumDecl.ooc
+++ b/source/rock/middle/EnumDecl.ooc
@@ -2,7 +2,7 @@ import structs/[ArrayList, HashMap]
 import ../io/TabbedWriter
 import TypeDecl, Declaration, Visitor, Node, VariableAccess, Type,
        VariableDecl, IntLiteral, FloatLiteral, Expression, FunctionDecl,
-       CoverDecl, Module, StructLiteral, BaseType
+       CoverDecl, Module, StructLiteral, BaseType, Version
 import tinker/[Trail, Resolver, Response, Errors]
 import ../frontend/Token
 
@@ -34,15 +34,16 @@ EnumDecl: class extends TypeDecl {
         }
 
         if (valuesCoverDecl == null) {
-            createCovers()
+            createCovers(trail)
             res wholeAgain(this, "need to resolve coverdecls for enum")
         }
 
         Response OK
     }
 
-    createCovers: func {
+    createCovers: func (trail: Trail) {
         valuesCoverDecl = CoverDecl new(name + "__values_t", token)
+
         for (v in getMeta() variables) {
             vDecl := VariableDecl new(BaseType new("Int", token), v name, v token)
             valuesCoverDecl addVariable(vDecl)
@@ -58,7 +59,21 @@ EnumDecl: class extends TypeDecl {
         slit := StructLiteral new(valuesCoverDecl getInstanceType(), elements, token)
         valuesGlobal = VariableDecl new(null, name + "__values", slit, token)
         valuesGlobal isGlobal = true
-        token module body add(valuesGlobal)
+
+        if (verzion) {
+            valuesCoverDecl verzion = verzion
+
+            // Find the version block upstream and add our globals and whatnot
+            for (i in 1 .. trail size + 1) {
+                match (trail peek(i)) {
+                    case vb: VersionBlock => if (vb spec == verzion) {
+                        vb body add(slit) . add(valuesGlobal)
+                    }
+                }
+            }
+        } else {
+            token module body add(valuesGlobal)
+        }
     }
 
     addFunction: func (fDecl: FunctionDecl) {

--- a/test/compiler/enums/versioned-enum.ooc
+++ b/test/compiler/enums/versioned-enum.ooc
@@ -1,0 +1,5 @@
+version (!gc) {
+    Foo: enum {
+        test = UNDEFINED_C_CONST
+    }
+}

